### ISSUE TITLE
Configurable option for Redirect query param name

### ISSUE THREAD

**Issue by mjrussell** [mjrussell/redux-auth-wrapper#16]

**Comment by joemcelroy** [comment]
hey, im interested in doing this issue. 

Could I add support to path to be able to handle both a string and a fn? 

```
  const UserIsAuthenticated = UserAuthWrapper({
    authSelector: (state) => {
      if (state.loginPage.token) {
        return { token: state.loginPage.token }
      }
    },
    redirectAction: routerActions.replace,
    wrapperDisplayName: 'UserIsAuthenticated',
    failureRedirectPath: (state) => {
       const {entity} = state.global
       return `${entity}`/login
    }
  })
```

**Comment by mjrussell** [comment]
@joemcelroy the original intent of this issue was to allow for redirect param names other than `redirect`. I.e. the auth-wrapper would redirect to `/login?myRedirect=foo` instead of `/login?redirect=foo`. That being said I'd be happy to accept a PR that adds support for failure redirect paths which are based on a function of state if that is useful for you

**Comment by joemcelroy** [comment]
@mjrussell sorry, I should of read the issue more closely. Great, I will submit a PR soon.

**Comment by qborreda** [comment]
I would love to have something configurable such as this, as my Routes all depend on the id passed in the first param of the tree .. 

Something like 
```
    <Route path="/" component={App}>
      <Route path="/:tenant_id" component={LoginPage} />
      <Route path="/:tenant_id/admin" component={UserIsAdmin(MainContainer)}>
        <IndexRoute component={DashboardPage} />
        <Route path="user-directory" component={UserDirectory}>
          <Route path=":userId" component={UserDirectory} />
        </Route>
      </Route>
    </Route>
```
.. and I would need to redirect to `/:tenant_id` whatever this id is ..

**Comment by mjrussell** [comment]
@joemcelroy are you working on the redirect parameter being allowed to also be a function of a redux selector? That would work for @qborreda because you could write:

```js
UserAuthWrapper({
  failureRedirectPath = (state, ownProps) => `${ownProps.params.tenant_id}/admin`,
  ...
})
```

Otherwise I can probably get a release out with this over the weekend

**Comment by mjrussell** [comment]
@joemcelroy @qborreda 

Just released v0.6.0 which adds support for both allowing `failureRedirectPath` to be a function as well as overriding the redirect query parameter.
